django_project/changes/tests/test_views.py

```diff
@@ -752,6 +752,349 @@ def test_VersionDelete_no_login(self):
+    def test_VersionListView_with_anonymouse_user(self):
+        """
+        Test if anonymous user can view version entry list.
+        """
+        response = self.client.get(reverse('version-list', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        expected_templates = [
+            'version/list.html', u'changes/version_list.html'
+        ]
+        self.assertEqual(
+            response.status_code, 200)
+        self.assertEqual(
+            response.template_name, expected_templates)
+        self.assertEqual(
+            response.context_data['object_list'][0],
+                         self.version)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionCreateView_with_anonymous_user(self):
+        """
+        Test if anonymous user can create a version entry.
+        """
+        response = self.client.get(reverse('version-create', kwargs={
+            'project_slug': self.project.slug
+        }))
+        self.assertEqual(
+            response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN=['testserver'])
+    def test_VersionUpdateView_with_anonymous_user(self):
+        """
+        Test if anonymous user can update a version entry.
+        """
+        response = self.client.get(reverse('version-create', kwargs={
+            'project_slug': self.project.slug
+        }))
+        self.assertEqual(
+            response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN=['testserver'])
+    def test_VersionDeleteView_with_anonymous_user(self):
+        """
+        Test if anonymous user can delete a version entry.
+        """
+        response = self.client.get(reverse('version-delete', kwargs={
+            'slug': self.version.slug,
+            'project_slug': self.version.project.slug
+        }))
+        self.assertEqual(
+            response.status_code, 302)
+
+
+class TestVersionViewsWithNormalUserForCRUD(TestCase):
+    """
+    Check if normal(a regular logged in user) user can perform CRUD
+    operations on version entries just in case they have the URL to the views.
+    """
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def setUp(self):
+        """
+        Setup before each test
+        We force the locale to en otherwise it will use
+        the locale of the host running the tests and we
+        will get unpredictable results / 404s
+        """
+
+        self.client = Client()
+        self.client.post(
+            '/set_language/', data={'language': 'en'})
+        logging.disable(logging.CRITICAL)
+        self.project = ProjectF.create(name='testproject')
+        self.version = VersionF.create(
+            project=self.project,
+            name='1.0.1')
+        self.category = CategoryF.create(
+            project=self.project,
+            name='testcategory')
+
+        self.user = UserF.create(**{
+            'username': 'sonlinux',
+            'password': 'password',
+        })
```

Comment:
To make it explicit, can you add 'staff: False'

Reply:
Hi @gustry staff is False by default in the UserF factory definition,
I think adding False would be showing the default.

Reply:
Yes, I know it's default, but sometimes, it's better to be explicit instead of implicit ;-)

A developer, in a few months, can quickly read and understand "this user is staff" when reading the code, instead of wondering if it's default.

Reply:
:-) Thanks, let me do so @Gustry